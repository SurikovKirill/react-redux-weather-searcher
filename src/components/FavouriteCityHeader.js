import React from 'react';
import '../containers/FavouriteContainer.css'

const FavouriteCityHeader = ({ onRemove, city }) => (
    <div className='favCityHeader'>
        {
            console.log(city.name)
        }
        <h1>{city.name}</h1>
        {!!city.img && <img src={city.img} alt='Иконка погоды' />}
        {!!city.temperature && <h1>{city.temperature}</h1>}
        <button className='deleteButton' onClick={() => onRemove(city.name)}>X</button>
    </div>
);


export default FavouriteCityHeader;