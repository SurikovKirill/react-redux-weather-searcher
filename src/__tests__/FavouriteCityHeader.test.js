import FavouriteCityHeader from '../components/FavouriteCityHeader';
import React from 'react';
import renderer from 'react-test-renderer';

describe('FavouriteCityHeader component', () => {

    test('matches the snapshot when city.icon != null, city.temperature != null', () => {
        const city = {
            name: 'Moscow',
            img: 'i',
            temperature: 't'
        };

        const tree = renderer.create(
            <FavouriteCityHeader city={city}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });


    test('matches the snapshot when city.icon == null, city.temperature == null', () => {
        const city = {
            name: 'Moscow',
            img: null,
            temperature: null
        };

        const tree = renderer.create(
            <FavouriteCityHeader city={city}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});