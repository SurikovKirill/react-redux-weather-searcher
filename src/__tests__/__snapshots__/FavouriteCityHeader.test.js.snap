// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteCityHeader component matches the snapshot when city.icon != null, city.temperature != null 1`] = `
<div
  className="favCityHeader"
>
  <h1>
    Moscow
  </h1>
  <img
    alt="Иконка погоды"
    src="i"
  />
  <h1>
    t
  </h1>
  <button
    className="deleteButton"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;

exports[`FavouriteCityHeader component matches the snapshot when city.icon == null, city.temperature == null 1`] = `
<div
  className="favCityHeader"
>
  <h1>
    Moscow
  </h1>
  <button
    className="deleteButton"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
